pyanaconda/modules/subscription/installation.py

```diff
@@ -0,0 +1,77 @@
+#
+# Copyright (C) 2020 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+import os
+
+from pyanaconda.core import util
+
+from pyanaconda.modules.common.task import Task
+from pyanaconda.modules.common.errors.installation import InsightsConnectError, \
+    InsightsClientMissingError
+
+from pyanaconda.anaconda_loggers import get_module_logger
+log = get_module_logger(__name__)
+
+
+class ConnectToInsightsTask(Task):
+    """Connect the target system to Red Hat Insights."""
+
+    INSIGHTS_TOOL_PATH = "/usr/bin/insights-client"
+
+    def __init__(self, sysroot, subscription_attached, connect_to_insights):
+        """Create a new task.
+
+        :param str sysroot: target system root path
+        :param bool subscription_attached: if True then the system has been subscribed,
+                                           False otherwise
+        :param bool connect_to_insights: if True then connect the system to Insights,
+                                         if False do nothing
+        """
+        super().__init__()
+        self._sysroot = sysroot
+        self._subscription_attached = subscription_attached
+        self._connect_to_insights = connect_to_insights
```

Comment:
Shouldn't these values be obtained dynamically? I'm thinking if they can't be different when the task is started.

Reply:
This being an installation task it should only be instantiated when Anaconda goes into the installation phase, at which point both of these should not change. At least if I understand all the machinery correctly. :)